### PR TITLE
Add doc section on component libraries

### LINKED ISSUE #70: Add topic on building component libraries
We should have a topic around the new component library template that shows how to build, package, and share libraries of components.

See https://blogs.msdn.microsoft.com/webdev/2018/04/17/blazor-0-2-0-release-now-available/ for some existing content.

### PR DESCRIPTION
**Pull request by stimms**
Adding section in the components page on using component libraries. Includes 

- How to create a library
- How to add it to solution
- How to consume it
- How to pack it for nuget

First docs commit so I'm very open to suggestions. 

Fixes #70

### CODE REVIEW

**Comment by guardrex** [comment]
Thanks @stimms for working on this.

@danroth27 Did you want this in a separate topic under the *Components* node? If so, I'll provide instructions to @stimms. If the content is fine in this topic, I'll perform a style+grammar+UE pass on it after your technical review.

File: docs/components/index.md [hunk 1]
```diff
@@ -598,3 +598,59 @@ The time is 10/04/2018 01:26:52.
...
+dotnet pack
+```
+
+The package can then be uploaded to NuGet using the [publish](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-push?tabs=netcore21) command
+
+```
+dotnet nuget publish
+```
+
+Any included static resources will be included in the NuGet package so consumers are not required to manually install them.
```
Comment (danroth27) [inline docs/components/index.md]: Consumers also don't need to add script or link tags for included JS and CSS files.  It's handled for you.

**Comment by danroth27** [comment]
Thanks for this contribution @stimms! The content looks great.

@guardrex  Yeah I think I would prefer this to be a top level node right under Components.

**Comment by guardrex** [comment]
@stimms I'll provide you instructions on how to set this up as a separate topic in a couple of hours.

**Comment by guardrex** [comment]
@stimms Ok ... back! ... Instructions ...

You can either:

* Close this PR. You'll create a new branch and a new PR (but obviously don't delete this branch until you have the new branch set up).
* Don't close this PR. Follow the instructions below and revert your changes to the Components topic at the end. **_This is probably the easiest way to go._**

1. Create a new topic named *component-libraries.md* in the *docs* folder.
1. Create a TOC entry for this in the *toc.md* file ...

   ```
   # [Component libraries](xref:client-side/blazor/component-libraries)
   ```
   ... it will become Line 5 just under the Components topic link.
1. At the top of the new topic, place the following metadata and content (modify the description as you see fit, but start it with an action verb, such as "discover" or "learn") ...

   ```
   ---
   title: Component libraries in Blazor
   author: stimms
   description: Discover how Blazor components can be included in a Blazor app from an external component library.
   manager: wpickett
   ms.author: riande
   ms.custom: mvc
   ms.date: 12/14/2018
   ms.prod: asp.net-core
   ms.technology: aspnet
   ms.topic: article
   uid: client-side/blazor/component-libraries
   ---
   # Component libraries in Blazor

   By [Simon Timms](https://github.com/stimms)

   [!INCLUDE[](~/includes/blazor-preview-notice.md)]

   START OF YOUR CONTENT HERE
   Begin with a short explanation and then use H2's (##) for sections thereafter.
   ```
1. Make Dan's [suggested update](#discussion_r241271471) :point_up:.
1. Revert your changes to the Components topic.
1. *Now push it! ... Ah, push it, push it real good!* ... :smile: &copy;1986 Salt-n-Pepa *Push It*

WRT a component library sample, we can discuss that, too. I don't think it's super necessary, but would be a *nice-to-have*. Let's discuss further if you're interested. If you'd prefer that I take care of that, no worries ... I can do it over the 🎅 holiday break 🎄.

**Comment by stimms** [comment]
Thanks for writing that all out @guardrex. I should have some time to get that squared away tomorrow. As for the sample I'm pretty sure I created one when I was writing this I just have to remember where I put it...
🕵️

**Comment by guardrex** [comment]
If you have (or create) a sample, then I need to give you additional info on where to put it and how to wire its code into the topic ....... or you can put it up in a GH repo for me to roll in later.

**Comment by guardrex** [comment]
@stimms If ur busy, I can take it from here if you like. Let me know.

**Comment by stimms** [comment]
This PR is superceded by  #343

**Pull request closed without merging**